models/product/src/generator.ts

```diff
@@ -0,0 +1,31 @@
+import { ClientLogging, Reference } from '@commercetools-test-data/commons';
+import { fake, Generator, oneOf } from '@commercetools-test-data/core';
+import * as ProductCatalogData from '@commercetools-test-data/product-catalog-data'; // TODO: fix import
+import { createRelatedDates } from '@commercetools-test-data/utils';
+import { productPriceMode } from './constants';
+import { TProduct } from './types';
+
+const [getOlderDate, getNewerDate] = createRelatedDates();
+
+// https://docs.commercetools.com/api/projects/products#product
+
+const generator = Generator<TProduct>({
+  fields: {
+    id: fake((f) => f.datatype.uuid()),
+    version: fake((f) => f.datatype.number()),
+    key: fake((f) => f.lorem.slug()),
+    productType: fake(() => Reference.random().typeId('product-type')),
```

Comment:
Should `productType`, `state`,  and `taxCategory` be the full representations and the fields build appropriate for the REST API (i.e. turned into references)?

Reply:
fixed with [9ba2456](https://github.com/commercetools/test-data/pull/219/commits/9ba2456baf09e23376964b590f05214d1d34ecaf) (thanks for all your help Steph!!!)